datalad/distribution/dataset.py

```diff
@@ -121,29 +100,25 @@ def _flyweight_id_from_args(cls, *args, **kwargs):
             path_ = cfg.obtain('datalad.locations.default-dataset')
         if path != path_:
             lgr.debug("Resolved dataset alias %r to path %r", path, path_)
+        return path_
 
-        # Sanity check for argument `path`:
-        # raise if we cannot deal with `path` at all or
-        # if it is not a local thing:
-        path_ = RI(path_).localpath
-
+    @classmethod
+    def _flyweight_postproc_path(cls, path):
         # we want an absolute path, but no resolved symlinks
-        if not isabs(path_):
-            path_ = opj(getpwd(), path_)
+        if not op.isabs(path):
+            path = op.join(op.getpwd(), path)
 
         # use canonical paths only:
-        path_ = normpath(path_)
-        kwargs['path'] = path_
-        return path_, args, kwargs
+        return op.normpath(path)
 
     def _flyweight_invalid(self):
         """Invalidation of Flyweight instance
 
-        Dataset doesn't need to be invalidated during its lifetime at all. Instead the underlying *Repo instances are.
+        Dataset doesn't need to be invalidated during its lifetime at all.
+        Instead the underlying *Repo instances are.
```

Comment:
When dealing with merge conflicts, please avoid this sort of unrelated formatting.  It introduces changes that aren't shared by any side, which are typically the changes of most interest, but here they're just noise.

Reply:
not 100% certain I did it consciously ... generally I am trying to avoid such changes while working on merges.  do you think it is worth rewriting for this one?

Reply:
That'd be my personal preference, but given what I perceive to be the preference of this project in general, I'd say "no, not if this were the only issue".  But if you agree that the `ok_file_under_git` point below should be addressed anyway, then I think it'd be good to resolve these minor issues in the same rewrite.